src/encoded/types/file.py

```diff
@@ -1191,14 +1229,101 @@ def download(context, request):
+            "ec": "Serverside File Download",       # Event Category
+            "ea": "Range Query" if request.range else "File Download", # Event Action
+            "el": filename,                         # Event Label
+            "ev": file_size_downloaded,             # Event Value
+            # Product fields
+            "pa": "purchase",
+            "ti": str(uuid4()),                     # We need to send a unique transaction id along w. 'transactions' like purchases
+            "pr1id": file_at_id,                    # Product ID/SKU
+            "pr1nm": filename,                      # Product Name
+            "pr1br" : lab.get("display_title"),     # Product Branch
+            "pr1qt": 1,                             # Product Quantity
+            # Product Category from @type, e.g. "File/FileProcessed"
+            "pr1ca": "/".join([ ty for ty in reversed(context.jsonld_type()[:-1]) ]),
+            # Product "Variant" (supposed to be like black, gray, etc), we repurpose for filetype for reporting
+            "pr1va": properties.get("file_type", "other") # "other" MATCHES THAT IN `file_type_detaild` calc property, since file_type_detailed is used on frontend when performing "Select All" files.
+        }
+
+        # Custom dimensions
+        # See https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#pr_cm_
+        if "name" in ga_config["dimensionNameMap"]:
+            ga_payload["dimension" + str(ga_config["dimensionNameMap"]["name"])] = filename
+            ga_payload["pr1cd" + str(ga_config["dimensionNameMap"]["name"])] = filename
+
+        if "experimentType" in ga_config["dimensionNameMap"]:
+            ga_payload["dimension" + str(ga_config["dimensionNameMap"]["experimentType"])] = file_experiment_type or None
+            ga_payload["pr1cd" + str(ga_config["dimensionNameMap"]["experimentType"])] = file_experiment_type or None
+
+        if "filesize" in ga_config["metricNameMap"]:
+            ga_payload["metric" + str(ga_config["metricNameMap"]["filesize"])] = file_size_downloaded
+            ga_payload["pr1cm" + str(ga_config["metricNameMap"]["filesize"])] = file_size_downloaded
+
+        if "downloads" in ga_config["metricNameMap"]:
+            ga_payload["metric" + str(ga_config["metricNameMap"]["downloads"])] = 0 if request.range else 1
+            ga_payload["pr1cm" + str(ga_config["metricNameMap"]["downloads"])] = 0 if request.range else 1
+
+
+        # client id (`cid`) or user id (`uid`) is required. uid shall be user uuid.
+        # client id might be gotten from Google Analytics cookie, but not stable to use and wont work on programmatic requests...
+        if user_uuid:
+            ga_payload['uid'] = user_uuid
+        if ga_cid:
+            ga_payload['cid'] = ga_cid
+
+        # TODO: WE SHOULD WRAP IN TRY/EXCEPT BLOCK RE: NETWORK?
+
+        # print('\n\n', ga_payload)
+
+        resp = requests.post(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'resp' is assigned to but never used](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=4885443)